pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonExtractIndexTransformFunction.java

```diff
@@ -108,8 +105,9 @@ public void init(List<TransformFunction> arguments, Map<String, ColumnContext> c
       _defaultValue = dataType.convert(((LiteralTransformFunction) fourthArgument).getStringLiteral());
     }
 
-    _resultMetadata = new TransformResultMetadata(dataType, true, false);
-    _matchingDocsMap = _jsonIndexReader.getMatchingDocsMap(_jsonPathString);
+    _resultMetadata = new TransformResultMetadata(dataType, isSingleValue, false);
+    _valueToMatchingFlattenedDocIdsMap =
+        _jsonIndexReader.getMatchingFlattenedDocsMap(inputJsonPath.substring(1));
```

Comment:
(minor) Suggest passing in the full path, and JSON index reader should validate the path